### PR TITLE
impl Error for TraceCtxError and ParseSpanIdError

### PR DESCRIPTION
**Pull request by Infrasonics**
By implementing `std::error::Error` it's possible to use them with `anyhow` for example. Also, from my understanding, it's expected behavior for errors anyway.

I tried to use a shorter, yet still understandable, version of the docstrings for the error messages. If you'd prefer something different, feel free to change or let me know what to put.

### CODE REVIEW

File: tracing-distributed/src/telemetry_layer.rs [hunk 1]
```diff
@@ -237,7 +237,7 @@ where
                         parent_id: Some(self.trace_ctx_registry.promote_span_id(parent_id)),
                         initialized_at,
                         meta: event.metadata(),
-                        service_name: &self.service_name,
```
Comment (jfhbrook-at-work) [inline tracing-distributed/src/telemetry_layer.rs:240]: This de-reference looks backwards incompatible. Is this change necessary to improve error handling?

File: tracing-distributed/src/trace.rs [hunk 2]
```diff
@@ -87,6 +87,21 @@ pub enum TraceCtxError {
     NoParentNodeHasTraceCtx,
 }
 
+impl std::fmt::Display for TraceCtxError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use TraceCtxError::*;
+        write!(f, "{}",
+            match self {
+                TelemetryLayerNotRegistered => "`TelemetryLayer` is not a registered subscriber of the current Span",
```
Comment (jfhbrook-at-work) [inline tracing-distributed/src/trace.rs:95]: These messages look OK to me, though I'd defer to @chrisdickinson for organizational opinions here.

**Review by jfhbrook-at-work — COMMENTED** [review]
I'm not so sure about the change to `telemetry_layer.rs` - is this needed? But I'm otherwise down to make these implement `Display` and `Error`!

@ericsampson do you think we can release these changes alongside #14 ? Or should these come out in separate releases? They seem like low-risk changes to me. Thoughts?

File: tracing-distributed/src/telemetry_layer.rs (hunk 1 above)
Comment (Infrasonics) [reply tracing-distributed/src/telemetry_layer.rs:240]: @jfhbrook-at-work This change is not necessary for the error handling. It's a warning that was only fixed since CI complained about it. I had not thought about backwards-compatibility, so thanks for  pointing this out.

As a warning: this same change is also in #12 for the same reason.

Comment (jfhbrook-at-work) [reply tracing-distributed/src/telemetry_layer.rs:240]: Thanks for the heads up! I'll keep an eye on it.

**Comment by jfhbrook-at-work** [comment]
Shipped in #14. Thanks!
